rfcs/rfc-create-ui-architecture.md

```diff
@@ -0,0 +1,146 @@
+- this is probably less work, even though we'll touch components often
+- can release new filing types as they are implemented
+
+### Recommendation
+
+Follow option 2.
+
+## Grooming Considerations
+
+1. should use a story per filing type (instead of a story per component)
+2. within stories, add tasks for different application steps/pages
+3. first story for a new filing type (other than IA) needs extra time to design and implement the resource/config object
+4. first story also needs extra time to create initial (parameterized) unit tests
+5. subsequent stories will need time to update the resources/config object
+6. subsequent stories will need time to update code where functionality differs between filing types
+7. subsequent stories will need time to update unit tests
+8. are there any dependencies? do we know what we need to, to begin?
+9. can we update a schema(s) so that the API can validate the "business logic" captured in the data? which other non-schema API validations will be needed?
+
+## Complications/Outstanding Questions
+
+1. we should investigate existing mechanisms/libraries to manage the resources object
+   - [vue-config](https://github.com/alex-oleshkevich/vue-config) may have code we can use to access a "config"
+   - see what we did for resources in Filings UI and maybe build upon it
+2. should we implement i18n for internationalization (ie, support for English/French/FN) at the same time?
+   - see [vue-i18n](https://kazupon.github.io/vue-i18n/)
+   - this is probably different enough that there is little value in doing it as the same time
+   - however, our resources/config object should support future internationalization (ie, whatever titles/blurbs it contains)
+3. should we add our resources/config object to the Vuex store?
+   - not if we don't have to (esp. if it's static data)
+4. should we support the fetching of "rules" like a date range from the API?
+   - either when the app starts, or as part of the data for the subject business/type
+   - inject those dynamically into the resources data (or store)?
+   - consider this when designing the resources architecture/data oject
+   - can be added/enhanced later
+5. what about different steps/routing between filing types?
+   - Create UI anticipated different steps but this is the first time we'd actually be configuring/using this and we haven't yet worked through the finer details (eg, overall filing validation)
+6. are there different entry points -- is there always a draft? other app init differences?
+7. may need to refactor existing data management (ie, Vuex store objects) to handle different filing types instead of what was developed specifically for IAs
+   - ie, generalize some data structures away from "IA" (and towards incorporation? continuation? generic filing?)
+8. how do we manage Feature Flags to deploy filing types independently?
+9. is there any new staff-only functionality? should this be configurable or hard-coded?
```

Comment:
Yes there is new staff-only functionality, this should be configurable

Reply:
Updated.